src/config.rs

```diff
@@ -0,0 +1,207 @@
+    Cyan,
+    White,
+}
+
+impl From<RawColour> for Colour {
+    fn from(raw_colour: RawColour) -> Colour {
+        match raw_colour {
+            RawColour::Black => Colour::Black,
+            RawColour::Red => Colour::Red,
+            RawColour::Green => Colour::Green,
+            RawColour::Yellow => Colour::Yellow,
+            RawColour::Blue => Colour::Blue,
+            RawColour::Purple => Colour::Purple,
+            RawColour::Cyan => Colour::Cyan,
+            RawColour::White => Colour::White,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+struct RawStyle {
+    pub foreground: Option<RawColour>,
+    pub background: Option<RawColour>,
+    pub underline: bool,
+    pub bold: bool,
+}
+
+impl Default for RawStyle {
+    fn default() -> RawStyle {
+        RawStyle{
+            foreground: None,
+            background: None,
+            underline: false,
+            bold: false,
+        }
+    }
+} // impl RawStyle
+
+impl From<RawStyle> for Style {
+    fn from(raw_style: RawStyle) -> Style {
+        let mut style = Style::default();
+
+        if let Some(foreground) = raw_style.foreground {
+            style = style.fg(Colour::from(foreground));
+        }
+
+        if let Some(background) = raw_style.background {
+            style = style.on(Colour::from(background));
+        }
+
+        if raw_style.underline {
+            style = style.underline();
+        }
+
+        if raw_style.bold {
+            style = style.bold();
+        }
+
+        style
+    }
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+struct RawConfig {
+    pub highlight_style: RawStyle,
+    pub description_style: RawStyle,
+    pub example_text_style: RawStyle,
+    pub example_code_style: RawStyle,
+    pub example_variable_style: RawStyle,
+}
+
+impl RawConfig {
+    fn new() -> RawConfig {
+        let mut raw_config = RawConfig::default();
+
+        raw_config.highlight_style.foreground = Some(RawColour::Red);
+        raw_config.example_variable_style.underline = true;
+
+        raw_config
+    }
+} // impl RawConfig
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct Config {
+    pub highlight_style: Style,
+    pub description_style: Style,
+    pub example_text_style: Style,
+    pub example_code_style: Style,
+    pub example_variable_style: Style,
+}
+
+impl From<RawConfig> for Config {
+    fn from(raw_config: RawConfig) -> Config {
+        Config{
+            highlight_style: Style::from(raw_config.highlight_style),
+            description_style: Style::from(raw_config.description_style),
+            example_text_style: Style::from(raw_config.example_text_style),
+            example_code_style: Style::from(raw_config.example_code_style),
+            example_variable_style: Style::from(raw_config.example_variable_style),
+        }
+    }
+}
+
+impl Config {
+    pub fn new() -> Result<Config, TealdeerError> {
```

Comment:
I would prefer not to call this method `new`, since I would not expect a call to `Config::new` to have the side effect of doing file system lookups.

What about renaming it to `load`?

Reply:
Sounds good.